src/main/haskell/kore/test/Test/Kore/Step/Function/Registry.hs

```diff
@@ -0,0 +1,274 @@
+                        (patternPureToKore
+                            (Top_ sortS
+                            :: CommonPurePattern Object)
+                        )
+                    }
+                ::KoreSentenceAxiom)
+            , asSentence
+                (SentenceAxiom
+                    { sentenceAxiomParameters = [asUnified sortVar]
+                    , sentenceAxiomAttributes = Attributes []
+                    , sentenceAxiomPattern =
+                        (patternPureToKore
+                            (And_ sortS (Top_ sortS)
+                                (And_ sortS (Top_ sortS)
+                                    (Rewrites_ sortS
+                                        (App_ fHead [])
+                                        (App_ tHead [])
+                                    )
+                                )
+                            :: CommonPurePattern Object)
+                        )
+                    }
+                ::KoreSentenceAxiom)
+            ]
+
+testIndexedModule :: KoreIndexedModule StepperAttributes
+testIndexedModule =
+    let
+        attributesVerification = defaultAttributesVerification Proxy
+        verifyResult = verifyAndIndexDefinition
+            attributesVerification
+            Builtin.koreVerifiers
+            testDef
+    in
+        case verifyResult of
+            Left err1            -> error (printError err1)
+            Right indexedModules ->
+                fromMaybe
+                    (error
+                        (  "The main module, '"
+                        ++ "test"
+                        ++ "', was not found. Check the --module flag."
+                        )
+                    )
+                    (Map.lookup (ModuleName "test") indexedModules)
+
+testId :: String -> Id level
+testId name =
+    Id
+        { getId = name
+        , idLocation = AstLocationTest
+        }
+
+testEvaluators
+    :: Map.Map (Id Object) [CommonApplicationFunctionEvaluator Object]
+testEvaluators = extractEvaluators Object testIndexedModule
+
+testMetadataTools :: MetadataTools Object StepperAttributes
+testMetadataTools = extractMetadataTools testIndexedModule
+
+test_functionRegistry :: [TestTree]
+test_functionRegistry =
+    [ testCase "Checking that two axioms are found for f"
+        (assertEqual ""
+            2
+            (length $ fromMaybe
+                (error "Should find precisely two axiom for f")
+                (Map.lookup (testId "f") testEvaluators)
+            )
+        )
+     , testCase "Checking that evaluator map has size 2"
+        (assertEqual ""
+            2
+            (Map.size testEvaluators)
+        )
+    , testCase "Checking that the indexed module contains a rewrite axiom"
+        (assertEqual ""
+            (1::Int)
+            (length (koreIndexedModuleToAxiomPatterns Object testIndexedModule))
+        )
+    , testCase "Checking that evaluator simplifies correctly"
```

Comment:
Not sure this is the right place to put this test, but it's better to have it than not to have it, I guess.

Reply:
It's mostly for coverage;  If I didn't put it, then the evaluator part in a lookup would show as unevaluated.